### DISCUSSION TITLE
Octopus deploy deployment status

### DISCUSSION THREAD

**Discussion by BBanoth** [cake-build/cake#3825] [Q&A]
Hello, I am using Jenkins and Octopus as ops tools.

In my cake file, I am creating octopus release and waiting for the deployment to finish.
           1. How can I cancel the deployment in octopus when I abort the deploy in cake
           2. How can I get the octopus deployment status once it finishes (Success/Cancelled/Failed etc ……)

**Comment by augustoproiete** [comment]
@BBanoth Post the code that you're using to deploy the Octopus release along with the steps you're taking to abort the deployment and some of us might be able to help and/or suggest alternatives

**Reply by BBanoth** [reply]
@augustoproiete : Here is the code I am using in cake

```
var createReleaseSettings = new CreateReleaseSettings {
		Server = octopusUrl,
		ApiKey = octopusApiKey,
		ReleaseNumber = releaseVersion,
		DeploymentTimeout = TimeSpan.FromMinutes(deploymentTimeout),
		WaitForDeployment = waitForDeployment
	}
	
	if(!string.IsNullOrEmpty(channelName)){
		createReleaseSettings.Channel = channelName;
		createReleaseSettings.ReleaseNumber = $"{releaseVersion}-{channelName}";
	}

	if(!string.IsNullOrEmpty(deployTo.Trim())){
		createReleaseSettings.DeployToMultiple= deployTo.Split(',').Select(p => p.Trim()).ToArray();
	}

	// create release
	OctoCreateRelease(projectName.Trim(), createReleaseSettings);
	Console.WriteLine($"***Create Octopus Release Deploy: finished");
```

Now I need to add try catch and know the deployment status(Success/Cancelled/Failed etc ……). How can I do that ?
And how can I cancel the deployment from cake script ?
